GitUI/GitUICommands.cs

```diff
@@ -701,93 +701,53 @@ bool Action()
 
         private bool StartResetChangesDialog(string fileName)
         {
             // Show a form asking the user if they want to reset the changes.
-            FormResetChanges.ActionEnum resetAction = FormResetChanges.ShowResetDialog(null, true, false);
+            FormResetChanges.ActionEnum resetType = FormResetChanges.ShowResetDialog(null, hasExistingFiles: true, hasNewFiles: false);
 
-            if (resetAction == FormResetChanges.ActionEnum.Cancel)
+            if (resetType == FormResetChanges.ActionEnum.Cancel)
             {
                 return false;
             }
 
             using (WaitCursorScope.Enter())
             {
                 // Reset all changes.
-                Module.ResetFile(fileName);
+                if (string.IsNullOrWhiteSpace(fileName))
+                {
+                    return Module.ResetAllChanges(resetAndDelete: resetType == FormResetChanges.ActionEnum.ResetAndDelete);
+                }
 
-                // Also delete new files, if requested.
-                if (resetAction == FormResetChanges.ActionEnum.ResetAndDelete)
+                string filePath = Path.GetRelativePath(Module.WorkingDir, fileName).ToPosixPath();
+                List<GitItemStatus> selectedItems = Module.GetAllChangedFilesWithSubmodulesStatus().Where(item => item.Name == filePath).ToList();
+                if (selectedItems.Count < 1)
                 {
-                    string? errorCaption = null;
-                    string? errorMessage = null;
-                    string? path = _fullPathResolver.Resolve(fileName);
-                    if (File.Exists(path))
-                    {
-                        try
-                        {
-                            File.Delete(path);
-                        }
-                        catch (Exception ex)
-                        {
-                            errorCaption = TranslatedStrings.ErrorCaptionFailedDeleteFile;
-                            errorMessage = ex.Message;
-                        }
-                    }
-                    else
-                    {
-                        errorCaption = TranslatedStrings.ErrorCaptionFailedDeleteFolder;
-                        path.TryDeleteDirectory(out errorMessage);
-                    }
+                    return false;
+                }
 
-                    if (errorMessage is not null)
-                    {
-                        MessageBox.Show(null, errorMessage, errorCaption, MessageBoxButtons.OK, MessageBoxIcon.Error);
-                    }
+                Module.ResetChanges(selectedItems, resetAndDelete: resetType == FormResetChanges.ActionEnum.ResetAndDelete, _fullPathResolver, out List<string> filesInUse, out StringBuilder output);
+                if (!string.IsNullOrWhiteSpace(output.ToString()))
+                {
+                    MessageBox.Show(null, output.ToString(), TranslatedStrings.ResetChangesCaption, MessageBoxButtons.OK, MessageBoxIcon.Error);
```

Comment:
Multiple calls to `output.ToString()`. Please use [`StringBuilder.Length` property](https://learn.microsoft.com/dotnet/api/system.text.stringbuilder.length) to check.

Reply:
ok